src/lib.rs

```diff
@@ -14,6 +14,7 @@ use diagnostics::{Diagnostics, DiagnosticsConfig, ERR_INDENT_SIZE};
 use net::hvm_to_net::hvm_to_net;
 
 pub mod diagnostics;
+#[allow(clippy::mutable_key_type)]
```

Comment:
Why is this needed here in the `mod` declaration?

Reply:
To allow this warning inside the `fun` module, the only place we are using HashSet/HashMap

Reply:
Shouldn't this be placed in wherever we use the hashmap with mutable key?

Reply:
Oh is it because it's too many places? in every single HashMap<Name>

Reply:
Yes